ops/charm.py

```diff
@@ -1273,16 +1294,33 @@ def __init__(self, raw: Optional[Dict[str, Any]] = None,
         self.name = raw_.get('name', '')
         self.summary = raw_.get('summary', '')
         self.description = raw_.get('description', '')
+        # The metadata spec says that these should be display-name <email>
+        # (roughly 'name-addr' from RFC 5322). However, many charms have only
+        # an email, or have a URL, or something else, so we leave these as
+        # a plain string.
         self.maintainers: List[str] = []
+        # Note that metadata v2 only defines 'maintainers' not 'maintainer'.
         if 'maintainer' in raw_:
             self.maintainers.append(raw_['maintainer'])
         if 'maintainers' in raw_:
             self.maintainers.extend(raw_['maintainers'])
+        if 'links' in raw_ and 'contact' in raw_['links']:
+            # When running tests, this might be loading from charmcraft.yaml.
+            self.maintainers.append(raw_['links']['contact'])
+        self._load_links(raw_)
+        # Note that metadata v2 does not define tags.
         self.tags = raw_.get('tags', [])
         self.terms = raw_.get('terms', [])
+        # Note that metadata v2 does not define series.
         self.series = raw_.get('series', [])
         self.subordinate = raw_.get('subordinate', False)
+        self.assumes = JujuAssumes.from_list(raw_.get('assumes', []))
+        # Note that metadata v2 does not define min-juju-version ('assumes'
+        # should be used instead).
         self.min_juju_version = raw_.get('min-juju-version')
+        if self.min_juju_version:
+            # Add in an implied 'assumes'.
+            self.assumes.features.append(f'juju >= {self.min_juju_version}')
```

Comment:
Are we sure we want to add this in? Why? It would seem better to me to have this reflect the metadata as is, rather than try to be smart about things.

Reply:
I definitely agree that the code I had earlier that tried to be smart in the other direction (populating `min_juju_version` from `assumes`) was a bad idea (on the positive side, it helped me really dig into how `assumes` works, so I could be more sure I had the class correct :smile:).

This direction (populating `assumes` from `min_juju_version`) is vastly simpler, just these 2-3 lines, and more obviously correct.

My thinking was that if someone is using this in code, and they want to support both versions of metadata (e.g. in a lib where you don't have as much control over the metadata), they have to do something like:

```python
min_version = meta.min_juju_version or find_minimum_juju_version(meta.assumes)
```

instead of just `find_minimum_juju_version(meta.assumes)`.

However, writing this out I think I have rubber-ducked my way into agreeing that it's too small to be worthwhile :smile:.